model_analyzer/result/run_result.py

```diff
@@ -57,33 +80,111 @@ def add_measurement(self, measurement):
+            of measurements in this RunResult
         """
 
         return self._measurements
 
+    def passing_measurements(self):
+        """
+        Returns
+        -------
+        list
+            of passing measurements in this RunResult
+        """
+
+        return self._passing_measurements
+
+    def failing_measurements(self):
+        """
+        Returns
+        -------
+        list
+            of failing measurements in this RunResult
+        """
+
+        return self._failing_measurements
+
     def top_n_measurements(self, n):
         """
         Parameters
         ----------
         n : int
-            The number of  top results 
+            The number of  top measurements 
             to retrieve
         
         Returns
         -------
-        The top n measurements in
-        this result
+        list os Measurements
+            The top n measurements in
+            this result
+        """
+
+        if len(self._passing_measurements) == 0:
+            logging.warn(
+                f"Requested top {n} measurements, but none satisfied constraints."
+                "Showing available constraint failing measurements for this config."
+            )
+
+            if n > len(self._failing_measurements):
+                logging.warn(
+                    f"Requested top {n} failing measurements, "
+                    f"but found only {len(self._failing_measurements)}."
+                    "Showing all available constraint failing measurements for this config."
+                )
+            return heapq.nsmallest(min(n, len(self._failing_measurements)),
+                                   self._failing_measurements)
+
+        if n > len(self._passing_measurements):
+            logging.warn(f"Requested top {n} measurements, but "
+                         f"found only {len(self._passing_measurements)}. "
+                         "Showing all available measurements for this config.")
+
+        return heapq.nsmallest(min(n, len(self._passing_measurements)),
+                               self._passing_measurements)
+
+    def instance_group_string(self):
+        """
+        Returns
+        -------
+        str
+            representation of the instance group used 
+            to generate this result
+        """
+        instance_group_list = self.run_config().model_config().get_config(
+        )['instance_group']
+        group_str_list = [
+            f"{group['count']}-{group['kind'].split('_')[1]}"
+            for group in instance_group_list
+        ]
+        return ','.join(group_str_list)
+
+    def dynamic_batching_string(self):
+        """
+        Returns
+        -------
+        str
+            representation of the dynamic batcher
+            configuration used to generate this result
         """
 
-        return heapq.nsmallest(n, self._measurements)
+        model_config = self.run_config().model_config()
+        if 'dynamic_batching' in model_config.get_config():
```

Comment:
If the config has dynamic batching but it does not have preferred batch size I think there will be issues.